src/tests/end-to-end/tests/details-view/headings.test.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Browser } from '../../common/browser';
+import { launchBrowser } from '../../common/browser-factory';
+import { DetailsViewCommonSelectors, GuidanceContentSelectors } from '../../common/element-identifiers/common-selectors';
+import { detailsViewSelectors } from '../../common/element-identifiers/details-view-selectors';
+import { popupPageElementIdentifiers } from '../../common/element-identifiers/popup-page-element-identifiers';
+import { enableHighContrast } from '../../common/enable-high-contrast';
+import { Page } from '../../common/page';
+import { scanForAccessibilityIssues } from '../../common/scan-for-accessibility-issues';
+
+describe('Headings Page', () => {
+    describe('Normal mode', () => {
+        let browser: Browser;
+        let targetTabId: number;
+        let headingsPage: Page;
+
+        beforeAll(async () => {
+            browser = await launchBrowser({ suppressFirstTimeDialog: true });
+            targetTabId = (await browser.setupNewTargetPage()).tabId;
+            headingsPage = await openHeadingsPage(browser, targetTabId);
+        });
+
+        afterAll(async () => {
+            if (browser) {
+                await browser.close();
+                browser = undefined;
+            }
+        });
+
+        it('should pass accessibility validation', async () => {
+            const results = await scanForAccessibilityIssues(headingsPage, GuidanceContentSelectors.detailsContent);
+            expect(results).toHaveLength(0);
+        });
+    });
+
+    describe('High contrast mode', () => {
+        let browser: Browser;
+        let targetTabId: number;
+        let headingsPage: Page;
+
+        beforeAll(async () => {
+            browser = await launchBrowser({ suppressFirstTimeDialog: true });
+
+            targetTabId = (await browser.setupNewTargetPage()).tabId;
+            await setupHighContrastMode();
+
+            headingsPage = await openHeadingsPage(browser, targetTabId);
+        });
+
+        afterAll(async () => {
+            if (browser) {
+                await browser.close();
+                browser = undefined;
+            }
+        });
+
+        it('should pass accessibility validation', async () => {
+            const results = await scanForAccessibilityIssues(headingsPage, GuidanceContentSelectors.detailsContent);
+            expect(results).toHaveLength(0);
+        });
+
+        async function setupHighContrastMode(): Promise<void> {
```

Comment:
Similarly, I think this logic is repeated now; maybe add a second variant of enableHighContrast that takes a browser instead of a page?

Reply:
Same reasoning as other comment; prefer to do in separate PR.